app/src/interfaces/select-multiple-checkbox-tree/select-multiple-checkbox-tree.vue

```diff
@@ -65,12 +71,22 @@ export default defineComponent({
 		},
 	},
 	emits: ['input'],
-	setup() {
+	setup(props) {
 		const { t } = useI18n();
 		const search = ref('');
-
+		const showSearch = ref(false);
 		const showSelectionOnly = ref(false);
 
+		onMounted(() => {
+			setTimeout(() => {
```

Comment:
Perhaps this can be watcher instead? `onMounted` and 100 ms `setTimeout` pairing feels a bit arbitrary here.

Reply:
I tried watcher first, but thing is I only want to have it update on the initial Mount, and not whenever value change. Secondly when just using onmounted the value prop was empty even when using nextTick. And thus settimeout

Reply:
ahh thanks for clarifying. Still feel a tad odd, but understandable. Wish something like [watchOnce](https://vueuse.org/shared/watchonce/) is builtin!

Reply:
I don't think this is the right solution for that use case. The 100ms is arbitrary and even based on browser performance in this case (it could even be that it will take longer than that 100ms). I believe `watch` returns a stop handle, so you could do something like:

```js
const stop = watch(() => props.choices, () => {
  // Do whatever you need to do
  stop();
}, { immediate: true });
```

to get the "watch once" behavior 🙂

Reply:
Here is what I did try but it didn't work:

```
const once = watch(
	props,
	({ choices, value }) => {
		nextTick(() => {
			const total = deepCount(choices);
			showSearch.value = total > 10;
			if (showSearch.value && (value?.length ?? false)) {
				showSelectionOnly.value = true;
			}
			once();
		});
	},
	{ immediate: true }
);
```